incubator/hnc/api/v1alpha2/hierarchy_types.go

```diff
@@ -155,137 +135,94 @@ type HierarchyConfigurationList struct {
+	// This may be an empty string.
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=32768
+	Message string `json:"message" protobuf:"bytes,6,opt,name=message"`
+}
 
-	// Affects is a list of group-version-kind-namespace-name that uniquely identifies
-	// the object(s) affected by the condition.
-	Affects []AffectedObject `json:"affects,omitempty"`
+// CreateCondition fills some required field with default values for schema
+// validation, e.g. Status and LastTransitionTime.
+func CreateCondition(tp, reason, msg string) Condition {
+	return Condition{
+		Type:   tp,
+		Status: "True",
+		// Set time as an obviously wrong value 1970-01-01T00:00:00Z since we
+		// overwrite conditions every time.
+		LastTransitionTime: metav1.Unix(0, 0),
+		Reason:             reason,
+		Message:            msg,
+	}
 }
 
 func (c Condition) String() string {
-	affects := fmt.Sprint(c.Affects)
-	msg := c.Msg
+	msg := c.Message
 	if len(msg) > 100 {
 		msg = msg[:100] + "..."
 	}
-	return fmt.Sprintf("%s: %s (affects %s)", c.Code, msg, affects)
-}
-
-// AffectedObject defines uniquely identifiable objects.
-type AffectedObject struct {
-	Group     string `json:"group,omitempty"`
-	Version   string `json:"version,omitempty"`
-	Kind      string `json:"kind,omitempty"`
-	Namespace string `json:"namespace,omitempty"`
-	Name      string `json:"name,omitempty"`
-}
-
-func NewAffectedNamespace(ns string) AffectedObject {
-	return AffectedObject{
-		Version: "v1",
-		Kind:    "Namespace",
-		Name:    ns,
-	}
-}
-
-func NewAffectedObject(gvk schema.GroupVersionKind, ns, nm string) AffectedObject {
-	return AffectedObject{
-		Group:     gvk.Group,
-		Version:   gvk.Version,
-		Kind:      gvk.Kind,
-		Namespace: ns,
-		Name:      nm,
-	}
-}
-
-// String should only be used for debug purposes
-func (a AffectedObject) String() string {
-	// No affected object (i.e. affects this namespace?). Note that this will never be returned by the
-	// API, but it is used internally to indicate that the API doesn't need to show an affected
-	// object.
-	if a.Name == "" {
-		return "<local>"
-	}
-
-	// No namespace -> it *is* a namespace
-	if a.Namespace == "" {
-		return a.Name
-	}
-
-	// Generic object (note that Group may be empty for core objects, don't worry about it)
-	return fmt.Sprintf("%s/%s/%s/%s/%s", a.Group, a.Version, a.Kind, a.Namespace, a.Name)
-}
-
-func SortAffectedObjects(objs []AffectedObject) {
-	sort.Slice(objs, func(i, j int) bool {
-		if objs[i].Group != objs[j].Group {
-			return objs[i].Group < objs[j].Group
-		}
-		if objs[i].Version != objs[j].Version {
-			return objs[i].Version < objs[j].Version
-		}
-		if objs[i].Version != objs[j].Version {
-			return objs[i].Version < objs[j].Version
-		}
-		if objs[i].Namespace != objs[j].Namespace {
-			return objs[i].Namespace < objs[j].Namespace
-		}
-		return objs[i].Name < objs[j].Name
-	})
+	return fmt.Sprintf("%s: %s: %s", c.Type, c.Reason, msg)
```

Comment:
Nit: "Type (reason): message" might be more readable than "Type: reason: message"